File: boilerplate/frontend/shared/recipes/all_auth.ts
import SuperTokens from "supertokens-web-js";
import Session from "supertokens-web-js/recipe/session";
import { defaultAppInfo, defaultRedirectionCallback, defaultOAuthProviders } from "../config/base";

export const SuperTokensConfig = {
    appInfo: {
        appName: defaultAppInfo.appName,
        apiDomain: defaultAppInfo.apiDomain,
        websiteDomain: defaultAppInfo.websiteDomain,
        apiBasePath: defaultAppInfo.apiBasePath,
        websiteBasePath: defaultAppInfo.websiteBasePath,
    },
    recipeList: [Session.init()],
    getRedirectionURL: defaultRedirectionCallback,
};

export const recipeDetails = {
    docsLink: "https://supertokens.com/docs/guides",
};

export function initSuperTokensUI() {
    (window as any).supertokensUIInit("supertokensui", {
        ...SuperTokensConfig,
        recipeList: [
            (window as any).supertokensUIEmailPassword.init(),
            (window as any).supertokensUIThirdParty.init({
                signInAndUpFeature: {
                    providers: [
                        (window as any).supertokensUIThirdParty.Github.init(),
                        (window as any).supertokensUIThirdParty.Google.init(),
                        (window as any).supertokensUIThirdParty.Apple.init(),
                        (window as any).supertokensUIThirdParty.Twitter.init(),
                    ],
                },
            }),
            (window as any).supertokensUIPasswordless.init({
                contactMethod: "EMAIL_OR_PHONE",
            }),
            (window as any).supertokensUISession.init(),
        ],
    });
}

export function initSuperTokensWebJS() {
    SuperTokens.init(SuperTokensConfig);
}

File: boilerplate/backend/go/http/config/all_auth.go
package main

import (
	"github.com/supertokens/supertokens-golang/recipe/dashboard"
	"github.com/supertokens/supertokens-golang/recipe/session"
	"github.com/supertokens/supertokens-golang/recipe/emailpassword"
	"github.com/supertokens/supertokens-golang/recipe/thirdparty"
	"github.com/supertokens/supertokens-golang/recipe/thirdparty/tpmodels"
	"github.com/supertokens/supertokens-golang/recipe/passwordless"
	"github.com/supertokens/supertokens-golang/recipe/passwordless/plessmodels"
	"github.com/supertokens/supertokens-golang/supertokens"
	sharedConfig "../../../shared/config"
)

var SuperTokensConfig = supertokens.TypeInput{
	Supertokens: &supertokens.ConnectionInfo{
		ConnectionURI: sharedConfig.DefaultSupertokensConfig["connectionURI"],
	},
	AppInfo: supertokens.AppInfo{
		AppName:       sharedConfig.DefaultAppInfo["appName"],
		APIDomain:     sharedConfig.DefaultAppInfo["apiDomain"],
		WebsiteDomain: sharedConfig.DefaultAppInfo["websiteDomain"],
	},
	RecipeList: []supertokens.Recipe{
		emailpassword.Init(nil),
		passwordless.Init(plessmodels.TypeInput{
			FlowType: "USER_INPUT_CODE_AND_MAGIC_LINK",
			ContactMethodEmailOrPhone: plessmodels.ContactMethodEmailOrPhoneConfig{
				Enabled: true,
			},
		}),
		thirdparty.Init(&tpmodels.TypeInput{
			SignInAndUpFeature: tpmodels.TypeInputSignInAndUp{
				Providers: []tpmodels.TypeProvider{
					{
						Config: tpmodels.ProviderConfig{
							ThirdPartyId: "google",
							Clients: []tpmodels.ProviderClient{
								{
									ClientID:     sharedConfig.DefaultOAuthProviders["google"].ClientID,
									ClientSecret: sharedConfig.DefaultOAuthProviders["google"].ClientSecret,
								},
							},
						},
					},
					{
						Config: tpmodels.ProviderConfig{
							ThirdPartyId: "github",
							Clients: []tpmodels.ProviderClient{
								{
									ClientID:     sharedConfig.DefaultOAuthProviders["github"].ClientID,
									ClientSecret: sharedConfig.DefaultOAuthProviders["github"].ClientSecret,
								},
							},
						},
					},
					{
						Config: tpmodels.ProviderConfig{
							ThirdPartyId: "apple",
							Clients: []tpmodels.ProviderClient{
								{
									ClientID:         sharedConfig.DefaultOAuthProviders["apple"].ClientID,
									AdditionalConfig: sharedConfig.DefaultOAuthProviders["apple"].Additional,
								},
							},
						},
					},
					{
						Config: tpmodels.ProviderConfig{
							ThirdPartyId: "twitter",
							Clients: []tpmodels.ProviderClient{
								{
									ClientID:     sharedConfig.DefaultOAuthProviders["twitter"].ClientID,
									ClientSecret: sharedConfig.DefaultOAuthProviders["twitter"].ClientSecret,
								},
							},
						},
					},
				},
			},
		}),
		session.Init(nil), // initializes session features
		dashboard.Init(nil),
	},
}
